minicbor-derive/src/encode.rs

```diff
@@ -137,11 +153,27 @@ fn on_enum(inp: &mut syn::DeriveInput) -> syn::Result<proc_macro2::TokenStream>
             syn::Fields::Named(f) if index_only => {
                 return Err(syn::Error::new(f.span(), "index_only enums must not have fields"))
             }
+            syn::Fields::Named(_) if flat => {
+                let (tests, statements) = encode_fields(&fields, false, encoding, true)?;
+                let idents = fields.fields().idents();
+                quote! {
+                    #name::#con{#(#idents,)* ..} => {
+                        #tests
+                        // Adding 2 to get size (enum index and 0-based indexing).
+                        let __size777 = __max_index777.unwrap_or_default() as u64 + 2;
```

Comment:
This is not correct if there are no fields. The array length would be 2 but contains only one element (the constructor index).